blockchain/txvalidator.go

```diff
@@ -84,14 +83,23 @@ func (b *BlockChain) CheckTransactionSanity(blockHeight uint32, txn *Transaction
 	return Success
 }
 
-// CheckTransactionContext verifys a transaction with history transaction in ledger
+// CheckTransactionContext verifies a transaction with history transaction in ledger
 func (b *BlockChain) CheckTransactionContext(blockHeight uint32, txn *Transaction) ErrCode {
 	// check if duplicated with transaction in ledger
 	if exist := b.db.IsTxHashDuplicate(txn.Hash()); exist {
 		log.Warn("[CheckTransactionContext] duplicate transaction check failed.")
 		return ErrTransactionDuplicate
 	}
 
+	// check to ensure only producer related tx existing in current block
+	// when inactive mode is on
+	if DefaultLedger.Arbitrators.IsInactiveMode() && !txn.IsProducerRelatedTx() &&
```

Comment:
Are InactiveArbitrators and IllegalType not a ProducerRelated transaction?
The comments saying only producer related tx is allowed, shouldn't it be
```go
if DefaultLedger.Arbitrators.IsInactiveMode() && !txn.IsProducerRelatedTx() {
    ...
}
```

Reply:
this comment should be "producer related tx and DPOS related tx", changed it.